### ISSUE TITLE
Checksum entire library

### ISSUE THREAD

**Issue by jojo2357** [DocDrydenn/kiwix-zim#14]
In continuation of the conversation on #13, I am opening this issue since I currently have an implementation of checksumming the user's library without an entire rewrite.

@Jaifroid @DocDrydenn when you get the chance, please clone and test [my fork](https://github.com/jojo2357/kiwix-zim) with the new `-f|--verfiy-library` flags. In dry run, no files will be deleted except for temp sha256 files. Going out of dry run *should* delete corrupt files, and we should consider what to do: redownload or just purge?

**Comment by Jaifroid** [comment]
I haven't tested yet, but how does this implementation deal with archives that are in the user's library but are no longer available from download.kiwix.org (a fairly common scenario, as the latter only keeps the last two of each archive)? Does it just ignore such archives?

**Comment by jojo2357** [comment]
Nothing happens as the archive is skipped just like in the regular process.

**Comment by jojo2357** [comment]
Yes, what I have done is when the user asks for a library check, it searches not for the newest zim, but for the zim in the library online. I *could* make it update as well, but my idea here was to just checksum, in case the user wants to hold onto an older version they can get the checksums checked without having their files updated.

In this case, purging happens as soon as a checksum fails (you can force this by renaming a zim to one version behind. I was using `wikipedia_ab_all_maxi_2023-0[34]` and you can see the behavior in the event of a checksum fail)

There wasnt really a way for me to delay purging given the current framework, and I think that purging as a step should be removed (see #7) anyway.
